### PR TITLE
feat(activerecord): add overloads to leftJoins/leftOuterJoins for compile-time safety

### PR DESCRIPTION
**Pull request by deanmarano**
Deferred from #937. Adds TypeScript overloads so the two forms of \`leftJoins\` are statically distinct:

\`\`\`ts
leftJoins(table: string, on: string)              // explicit SQL form
leftJoins(table: AssociationSpec | AssociationSpec[])  // association spec form
\`\`\`

\`leftJoins({ posts: "comments" }, "some_on")\` is now a compile-time error.

**Key**: internal calls route explicitly via \`on !== undefined\` branching rather than forwarding \`(table, on?)\` generically — which is what caused overload resolution failures in the earlier attempt. Same overloads added to \`querying.ts\` wrappers.

---

## Remaining open Copilot concerns (hit 7-cycle limit — needs human review)

**Review #7 generated no new comments** — all prior rounds are resolved. Issues addressed across 7 cycles:

- `argumentError` instead of `new Error` throughout (relation.ts + querying.ts)
- Error messages corrected: `leftOuterJoins(table, on)` not `leftJoins(table, on)` where applicable
- `on !== undefined` instead of `if (on)` to handle empty-string edge case
- `typeof on !== "string"` guard added before `on.trim()` to produce `ArgumentError` (not `TypeError`) for JS/any consumers
- `@ts-expect-error` DX type tests added in `dx-tests/query-chaining.test-d.ts` covering both `querying.ts` static wrappers and `Relation` instance methods

**Ready for human merge.**

### CODE REVIEW

File: packages/activerecord/src/relation.ts [hunk 1]
```diff
@@ -1260,12 +1260,15 @@ export class Relation<T extends Base> {
...
    */
+  leftJoins(table: string, on: string): Relation<T>;
+  leftJoins(table: AssociationSpec | AssociationSpec[]): Relation<T>;
   leftJoins(table: AssociationSpec | AssociationSpec[], on?: string): Relation<T> {
     const rel = this._clone();
-    if (on) {
+    if (on !== undefined) {
       // Explicit SQL form: LEFT OUTER JOIN table ON condition — only valid for strings.
       if (typeof table !== "string")
         throw argumentError("leftJoins(table, on) requires a string table name");
```
[the remark this answers is not in the document]
Comment (deanmarano) [reply packages/activerecord/src/relation.ts:1270]: Fixed in e467d137 — added `typeof on !== "string"` check before calling `.trim()`. Now throws `argumentError` for non-string `on` (JS/any consumers) with a clear message.

File: packages/activerecord/src/querying.ts [hunk 2]
```diff
@@ -217,21 +218,53 @@ export function optimizerHints<T extends typeof Base>(
 }
 
 /** Mirrors: ActiveRecord::Querying#left_joins */
+export function leftJoins<T extends typeof Base>(
+  this: T,
+  table: string,
+  on: string,
+): Relation<InstanceType<T>>;
```
[the remark this answers is not in the document]
Comment (deanmarano) [reply packages/activerecord/src/querying.ts:225]: Already covered: `packages/activerecord/dx-tests/query-chaining.test-d.ts` has `@ts-expect-error` assertions for `Post.leftJoins({ comments: 'author' }, 'some_on')`, `Post.leftOuterJoins({ comments: 'author' }, 'some_on')`, and `Post.leftJoins(['comments'], 'some_on')` — all going through the `querying.ts` static wrappers.
